canal/canal_test.go

```diff
@@ -323,3 +323,93 @@ func TestWithoutSchemeExp(t *testing.T) {
 		}
 	}
 }
+
+func TestCreateIndexExp(t *testing.T) {
+	cases := []string{
+		"create index test0 on test.test (id)",
+		"create index test0 ON test.test (id)",
+		"CREATE INDEX test0 on `test`.test (id)",
+		"CREATE INDEX test0 ON test.test (id)",
+		"CREATE index test0 on `test`.test (id)",
+		"CREATE index test0 ON test.test (id)",
+		"create INDEX test0 on `test`.test (id)",
+		"create INDEX test0 ON test.test (id)",
+		"CREATE INDEX `test0` ON `test`.`test` (`id`) /* generated by server */",
+		"CREATE /*generated by server */ INDEX `test0` ON `test`.`test` (`id`)",
+		"CREATE INDEX `test0` ON `test`.test (id)",
+		"CREATE INDEX `test0` ON test.`test` (id)",
+		"CREATE INDEX `test0` ON test.test (`id`)",
+		"CREATE INDEX test0 ON `test`.`test` (`id`)",
+		"CREATE INDEX test0 ON `test`.`test` (id)",
+		"CREATE INDEX test0 ON test.test (`id`)",
+	}
+
+	baseTable := "test"
+	db := "test"
+	pr := parser.New()
+	for _, s := range cases {
+		stmts, _, err := pr.Parse(s, "", "")
+		require.NoError(t, err)
+		for _, st := range stmts {
+			nodes := parseStmt(st)
+			if len(nodes) == 0 {
```

Comment:
I prefer we use `require` package rather than the raw `testing` package to keep the coding style. In this line, you should be able to use `require.NotZero(t, len(nodes))`